openshift/README.md

```diff
@@ -7,57 +7,132 @@ This project uses OpenShift container platform (v3.11) for hosting the applicati
+3. Create a deploy.env file that contains the values for the parameters within the template
+4. Create Objects `oc process pims-api-deploy --param-file=[.env] | oc create -f -`
 
 # Environments
 
 There are 4 environments setup in the OpenShift Platform.
 
-- Tools : Contains all the tools/external services required by the application.
-- Dev : Contains a running application instance of an open pull request branch.
-- Test : Contains a running application instance of the current release branch.
-- Prod : Contains a running application instance of the current state of master.
+- **Tools** - Contains all the tools/external services required by the application.
+- **Dev** - Contains a running application instance of an open pull request branch.
+- **Test** - Contains a running application instance of the current release branch.
+- **Prod** - Contains a running application instance of the current state of master.
 
 # Filename Convention
 
 Files are named with the following convention in mind:
 
-- [APPLICATION]-build.yaml: Build configuration for the application. Used for building any images required by the deployment configuration.
-
-- [APPLICATION]-deploy.json: Deployment configuration for the application. These deployment pods are long lived.
+| Object                    | Description                                                                                                     |
+| ------------------------- | --------------------------------------------------------------------------------------------------------------- |
+| [APPLICATION]-build.yaml  | Build configuration for the application. Used for building any images required by the deployment configuration. |
+| [APPLICATION]-deploy.yaml | Deployment configuration for the application. These deployment pods are long lived.                             |
 
 # CI/CD Tools
 
 ## Jenkins
 
-Jenkins is being used to orchestrate jobs between Github and OpenShift. Anytime a PR is open against the release/master branch it triggers a job on Jenkins which runs the pipeline stages and reports the status back to Github.
-
-[Link to Documentation](https://docs.openshift.com/container-platform/3.10/using_images/other_images/jenkins.html)
-
-[Link to the jenkins s2i image](https://github.com/BCDevOps/openshift-components/tree/master/cicd/jenkins)
+Jenkins is being used to orchestrate jobs between GitHub and OpenShift. Anytime a Pull Request is opened against the release/master branch, it triggers a job on Jenkins which runs the pipeline stages and reports the status back to GitHub.
+
+This project makes use of several pipelines. For each one, use the generic pipeline template located at `/openshift/templates/tools/pipeline.yaml` and customize with appropriate values.
+
+### cicd-pipeline
+
+1. View the parameters `oc process --parameters -f openshift/templates/tools/pipeline.yaml`
+2. Create a **`cicd.env`** file that contains the values for the parameters within the template
+    ```
+    NAME=cicd
```

Comment:
Why does the Jenkinsfile need the extension `cicd`?

Reply:
We can name the Jenkinsfile however we want. In order to have multiple files for various pipelines we need to agree on a pattern for naming them:

I have seen 3 patterns to deal with this on other `bcgov` repos:

1. Use multiple (nested) folders, each one containing a single **Jenkinsfile**
    See https://github.com/bcgov/jag-cullencommission/tree/master/jenkins

2. Having them in a single folder but appending a suffix to indicate the purpose of the Jenkinsfile
    See https://github.com/bcgov/moh-prime
    They have: Jenkinsfile, Jenkinsfile.promotion, Jenkinsfile.tools

3. Having them in a single folder and rename them to *.groovy files, given that groovy is their syntax
     See https://github.com/bcgov/BCSC-SS/tree/dev/jenkins and https://github.com/bcgov/bcrs-business-create-ui/tree/master/Jenkinsfiles

I personally like to keep the reference to a Jenkinsfile-[whatever] vs having a "build.groovy" file. But if the team feels different I'm open to hearing ideas

Reply:
Ah I understand.  I'm fine with your approach, it makes sense.